lib/api.js

```diff
@@ -8,100 +8,93 @@ var sec = require('fh-security'),
+  // Main FHapi constructor function..
+  //
+  function FHapi() {
+    EventEmitter.call(this);
+    this.redisPort = cfg.redis.port || '6379';
+    this.redisHost = cfg.redis.host || 'localhost';
+    this.sec = sec.security;
+  }
+  util.inherits(FHapi, EventEmitter);
+
+  FHapi.prototype.getVersion = function getVersion(){
+    //Getting The Version of fh-mbaas-api
+    return packageJSON.version;
+  };
+  FHapi.prototype.cache = require('./cache')(cfg);
+  FHapi.prototype.db = require('./db')(cfg);
+  FHapi.prototype.forms = require('./forms')(cfg);
+  FHapi.prototype.push = require('./push')(cfg);
+  FHapi.prototype.call = require('./call')(cfg);
+  FHapi.prototype.session = require('./session')(cfg);
+  FHapi.prototype.stats = require('./stats')(cfg);
+  FHapi.prototype.sync = require('./sync').api;
+  FHapi.prototype.act = require('./act')(cfg);
+  FHapi.prototype.service = require('./act')(cfg);
+  FHapi.prototype.auth = require('./auth')(cfg);
+  FHapi.prototype.host = require('./host');
+  FHapi.prototype.permission_map = require('fh-db').permission_map;
+  FHapi.prototype.hash = function hash(opts, callback) {
+    var p = {
+      act: 'hash',
+      params: opts
+    };
+    sec.security(p, callback);
+  },
+  FHapi.prototype.mbaasExpress = function (opts) {
+    opts = opts || {};
+    opts.api = this;
+    return require('fh-mbaas-express')(opts);
+  };
+  FHapi.prototype.shutdown = function (cb) {
+    // Sync service has a setInterval loop running which will prevent fh-mbaas-api from exiting cleanly.
+    // Call stopAll to ensure Sync exits clenaly.
+    this.sync.stopAll(cb);
+  };
+  FHapi.prototype.setLogLevel = function(level) {
+    logr.setLogLevel(level);
   };
 
-  var redisUrl = 'redis://' + api.redisHost + ':' + api.redisPort;
+  var api = new FHapi();
 
+  // Getting the db connection string will do 2 things:
+  // * set the appropriate env var with a mongo connection url
+  // * allow sync to be initialised, as sync requires a direct connection to mongodb
   api.db({
     "act" : "connectionString"
   }, function(err, connectionString) {
+    var msg = 'Warning! Could not get a mongodb connection string. Sync will not work.';
     if (err) {
-      console.warn('Warning! Could not get a mongodb connection string. Sync will not work. (', err, ')');
+      console.warn(msg,' (', err, ')');
       return;
     } else if (!connectionString) {
-      console.warn('Warning! Could not get a mongodb connection string. Sync will not work. If running in a Dynofarm/FeedHenry MBaaS, ensure the database is upgraded');
+      console.warn(msg, 'If running in a Dynofarm/FeedHenry MBaaS, ensure the database is upgraded');
       return;
     }
+
+    var redisUrl = 'redis://' + api.redisHost + ':' + api.redisPort;
     api.sync.connect(connectionString, redisUrl, null, function(err) {
       if (err) {
         console.error('Error starting the sync server (', err, ')');
+        return;
       }
+      api.emit('sync_ready');
```

Comment:
@wei-lee Good point. After a successful `connect`, sync could emit it.
This would follow the pattern proposed by @evanshortiss in PR #80 where an event emitter is passed to sync, and that event emitter is initialised up at the fh-mbaas-api level